addons/mail/models/mail_message.py

```diff
@@ -816,6 +819,18 @@ def _message_remove_reaction(self, content):
         reaction.unlink()
         self.env[self.model].browse(self.res_id)._message_remove_reaction_after_hook(message=self, content=content)
 
+    def _message_cleanup_related_data(self):
+        """
+            Clean related message data.\n
+            As this method unlinks notifications record in sudo, be sure that access is granted and check that it won't have unwanted side effects.
+        """
+        for message in self:
+            if message.starred_partner_ids:
+                message.write({'starred_partner_ids': [(5, 0, 0)]})
+
+            if message.notification_ids:
```

Comment:
Accessing notification_ids depends on current user ACLs ... shouldn't we sudo previously in the loop , or directly call ``_message_cleanup_related_data`` as sudo ?